runtime/protocol/http-client-engines/http-client-engine-crt/common/src/aws/smithy/kotlin/runtime/http/engine/crt/RequestUtil.kt

```diff
@@ -34,17 +34,18 @@ internal val HttpRequest.uri: Uri
 internal fun HttpRequest.toCrtRequest(callContext: CoroutineContext): aws.sdk.kotlin.crt.http.HttpRequest {
     val body = this.body
     check(!body.isDuplex) { "CrtHttpEngine does not yet support full duplex streams" }
-    val bodyStream = when (body) {
-        is HttpBody.Empty -> null
-        is HttpBody.Bytes -> HttpRequestBodyStream.fromByteArray(body.bytes())
-        is HttpBody.ChannelContent -> ReadChannelBodyStream(body.readFrom(), callContext)
-        is HttpBody.SourceContent -> {
+    val bodyStream = when {
+        body is HttpBody.Empty || this.isChunked -> null
+        body is HttpBody.Bytes -> HttpRequestBodyStream.fromByteArray(body.bytes())
+        body is HttpBody.ChannelContent -> ReadChannelBodyStream(body.readFrom(), callContext)
+        body is HttpBody.SourceContent -> {
             val source = body.readFrom()
             callContext.job.invokeOnCompletion {
                 source.close()
             }
             SdkSourceBodyStream(source)
         }
+        else -> null
```

Comment:
**Question**: What could `else` be if not one of the above cases? Should the stream be `null` or should we actually throw an error?

Reply:
Probably throw an error. I wish there was a better way to format this `when` block, ideally it would enumerate all the `HttpBody` types and nothing else, but I needed access to the `HttpRequest` to call `isChunked`

Reply:
this should throw an error on the `else` clause now

Reply:
I refactored to get rid of the dangling `else`:

```kotlin
    val bodyStream = if (isChunked) null else when (body) {
        is HttpBody.Empty -> null
        ...
    }
```